lib/browser/api/web-contents.js

```diff
@@ -131,40 +122,18 @@ const asyncWebFrameMethods = function (requestId, method, callback, ...args) {
         if (typeof callback === 'function') callback(result)
         resolve(result)
       } else {
-        if (error.__ELECTRON_SERIALIZED_ERROR__ && errorConstructors[error.name]) {
-          const rehydratedError = new errorConstructors[error.name](error.message)
-          rehydratedError.stack = error.stack
-
-          reject(rehydratedError)
-        } else {
-          reject(error)
-        }
+        reject(errorUtils.deserialize(error))
       }
     })
   })
 }
 
-const syncWebFrameMethods = function (requestId, method, callback, ...args) {
-  this.send('ELECTRON_INTERNAL_RENDERER_SYNC_WEB_FRAME_METHOD', requestId, method, args)
-  ipcMain.once(`ELECTRON_INTERNAL_BROWSER_SYNC_WEB_FRAME_RESPONSE_${requestId}`, function (event, result) {
-    if (callback) callback(result)
-  })
-}
-
 for (const method of webFrameMethods) {
   WebContents.prototype[method] = function (...args) {
     this.send('ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', method, args)
```

Comment:
@alexeykuzmin, @MarshallOfSound do you think this change would make sense as well? Not sure if errors should be logged here or on the renderer side.
```
-    this.send('ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', method, args)
+    asyncWebFrameMethods.call(this, getNextId(), method, undefined, ...args)
+      .catch(error => {
+        console.error(`${error}`)
+      })
```

Handling of the `ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD` message would be removed from `web-frame-init.js`

Reply:
Is calling `console.error()` on the "browser" side the same as calling it on the "renderer" side?

Reply:
No, on browser side it prints to stdout, on renderer it goes to devtools.